SampleScripts/PowerManagement-ManagementDomain.ps1

```diff
@@ -186,12 +186,185 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
+                                            }
+
+                                            # Stop vSphere HA to avoid "orphaned" VMs during vSAN shutdown
+                                            if (Test-vSphereAuthentication -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass) {
+                                                if (!$(Set-VsphereHA -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -cluster $clustername -disableHA)) {
+                                                    Write-PowerManagementLogMessage -Type ERROR -Message "Could not disable vSphere High Availability for cluster '$clustername'. Exiting!"
+                                                }
+                                            }
+                                            # Check if there are VMs running on a vSAN HCI Mesh
+                                            if (Test-vSphereAuthentication -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass) {
+                                                $RemoteVMs = @()
+                                                $RemoteVMs = Get-poweronVMsOnRemoteDS $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -clustertocheck $clustername
+                                                if ($RemoteVMs.count -eq 0) {
+                                                    Write-PowerManagementLogMessage -Type INFO -Message "All remote VMs are powered off."
+                                                } else {
+                                                    Write-PowerManagementLogMessage -Type ERROR -Message "Some remote VMs are still powered-on : $($RemoteVMs.Name). Cannot proceed until the powered-on VMs are shut down. Check your environment."
+                                                }
+                                            }
+
+                                            #Testing VSAN health
+                                            if ( (Test-VsanHealth -cluster $clustername -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass) -eq 0) {
+                                                Write-PowerManagementLogMessage -Type INFO -Message "vSAN cluster health is good."
+                                            } else {
+                                                Write-PowerManagementLogMessage -Type WARNING -Message "The vSAN cluster isn't in a healthy state. Check the vSAN status in Cluster '$($clustername)'. After you resolve the vSAN issues, run the script again."
+                                                Write-PowerManagementLogMessage -Type WARNING -Message "If the script has reached ESXi vSAN shutdown previously, this error is expected. Continue by following the documentation of VMware Cloud Foundation. "
```

Comment:
```suggestion
                                                Write-PowerManagementLogMessage -Type WARNING -Message "If the script has previously reached ESXi shutdown, this warning is expected. Continue by following the VMware Cloud Foundation product documentation."
```

Do we have a refernece where to look for this documentation?

Reply:
Thought.... odd one.

If the version is known, could we do a read host for Y/N and if Y open the documentation link for the version?

Reply:
Will address during Code review